gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py

```diff
@@ -36,59 +55,38 @@ def _sim(self, x1, x2, postprocess, x1_eq_x2=False, metric='tanimoto'):
             (:class:`Tensor`, :class:`Tensor) corresponding to the similarity matrix between `x1` and `x2`
         """
 
-        # Extend this list when including new metrics
-
-        if metric not in ['tanimoto']:
-            raise RuntimeError('Similarity metric not supported. Available options are \'tanimoto\'')
-
         # Branch for Tanimoto metric
-        if metric == 'tanimoto':
-
-            # Tanimoto distance is proportional to (<x, y>) / (||x||^2 + ||y||^2 - <x, y>) where x and y are bit vectors
-
-            x1_norm = x1.pow(2).sum(dim=-1, keepdim=True)  # Compute the squared L2-norm of each datapoint
-            if x1_eq_x2 and not x1.requires_grad and not x2.requires_grad:
-                x2_norm = x1_norm
-            else:
-                x2_norm = x2.pow(2).sum(dim=-1, keepdim=False)
-
-            # The following code works for both the batch and non-batch cases
-            cross_product = x1.matmul(x2.transpose(-2, -1))
-
-            # Analogue of denominator in Tanimoto formula
-            denominator = x1_norm + x2_norm.squeeze(-1) - cross_product
-
-            res = (cross_product/denominator)
-
-            # if x1 and x2 are equal than the Tanimoto similarity is 1
-            if x1_eq_x2 and not x1.requires_grad and not x2.requires_grad:
-                res.diagonal(dim1=-2, dim2=-1).fill_(1)
-
-            # Zero out negative values
+        if metric == "tanimoto":
```

Comment:
So I guess we'll extend this with elif statements to support the other bit vector distances?

Reply:
That was my thought!